File: common-core/src/main/java/com/ityu/bean/entity/system/Role.java
package com.ityu.bean.entity.system;


import com.ityu.bean.entity.BaseEntity;
import lombok.Data;
import org.hibernate.annotations.Table;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.validation.constraints.NotBlank;

/**
 * Created  on 2018/4/2 0002.
 *
 * @author enilu
 */
@Entity(name = "t_sys_role")
@Table(appliesTo = "t_sys_role", comment = "角色")
@Data
public class Role extends BaseEntity {
    @Column
    private Integer num;
    @Column
    private Long pid;
    @Column
    @NotBlank(message = "角色名称不能为空")
    private String name;
    @Column
    private Long deptid;
    @Column
    private String tips;
    @Column
    private Integer version;

}

File: common-core/src/main/java/com/ityu/common/bean/entity/system/Role.kt
package com.ityu.common.bean.entity.system

import com.ityu.common.bean.entity.BaseEntity
import jakarta.persistence.Table
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Transient
import jakarta.validation.constraints.NotBlank

/**
 * Created  on 2018/4/2 0002.
 *
 * @author enilu
 */
@Entity(name = "t_sys_role")
@Table(name = "t_sys_role", )
open class Role : BaseEntity() {
    @Column(columnDefinition = "INT COMMENT '排序'")
    open var num: Int? = null

    @Column(columnDefinition = "BIGINT COMMENT '父角色id'")
    open var pid: Long? = null

    @Column(columnDefinition = "VARCHAR(64) COMMENT '角色名称'")
    open var name: @NotBlank(message = "角色名称不能为空") String? = null

    @Column(columnDefinition = "VARCHAR(64) COMMENT '角色编码'")
    open var code: String? = null

    @Column(columnDefinition = "BIGINT COMMENT '角色所属部门'")
    open var deptid: Long? = null

    @Column(columnDefinition = "INT COMMENT '角色版本号'")
    open var version: Int? = null

    @Transient
    open val label = name
}
